datalad/interface/utils.py

```diff
@@ -667,3 +667,84 @@ def _discover_trace_to_known(path, trace, spec):
         if not isdir(p):
             continue
         _discover_trace_to_known(p, trace, spec)
+
+
+def filter_unmodified(content_by_ds, refds, since):
+    """Filter per-dataset path specifications based on modification history.
+
+    This function takes a path specification dictionary, as produced by
+    `Interface._prep()` and filters it such that only that subset of paths
+    remains in the dictionary that correspdong to the set of changes in
+    the given reference dataset since a given state.
+
+    The change set is traced across all related subdatasets, i.e. if a submodule
+    in the reference dataset is reported as modified then all paths for any given
+    subdataset in the modified one are tested for changes too (based on the
+    state at which the parent dataset reports a change in the subdataset), and so
+    on.
+
+    In doing so, not only unmodified given paths are removed, but also modified
+    given paths are replaced by the set of actually modified paths within them.
+
+    Only commited changes are considered!
+
+    Parameters
+    ----------
+    content_by_ds : dict
+      Per-dataset path specifications, as produced ,for example, by
+      `Interface._prep()`
+    refds : Dataset
```

Comment:
my email reply didn't post... I was talking about recursive invocation within this function:
```
filter_unmodified(mod_subs, Dataset(subds_path), modified[subds_path])
```
so it would possibly create those datasets whenever there is no really any use for them since only repo is used.